File: typescript/src/tools/dashboards/getAll.ts
import { DashboardGetAllSchema } from "@/schema/tool-inputs";
import { getToolDefinition } from "@/tools/toolDefinitions";
import type { Context, Tool } from "@/tools/types";
import type { z } from "zod";

const schema = DashboardGetAllSchema;

type Params = z.infer<typeof schema>;

export const getAllHandler = async (context: Context, params: Params) => {
	const { data } = params;
	const projectId = await context.getProjectId();
	const dashboardsResult = await context.api
		.dashboards({ projectId })
		.list({ params: data ?? {} });

	if (!dashboardsResult.success) {
		throw new Error(`Failed to get dashboards: ${dashboardsResult.error.message}`);
	}

	return { content: [{ type: "text", text: JSON.stringify(dashboardsResult.data) }] };
};

const tool = (): Tool<typeof schema> => ({
	name: "dashboards-get-all",
	description: getToolDefinition("dashboards-get-all").description,
	schema,
	handler: getAllHandler,
});

export default tool;

File: python/tools/dashboards/get_all.py
import json

from api.client import is_error, is_success
from schema.tool_inputs import DashboardGetAllSchema
from tools.tool_definitions import get_tool_definition
from tools.types import Context, Tool, ToolResult


async def get_all_dashboards_handler(context: Context, params: DashboardGetAllSchema) -> ToolResult:
    project_id = await context.get_project_id()
    dashboards_result = await context.api.dashboards(project_id).list(params.data)

    if is_error(dashboards_result):
        raise Exception(f"Failed to get dashboards: {dashboards_result.error}")

    assert is_success(dashboards_result)

    dashboards_data = [dashboard.model_dump() for dashboard in dashboards_result.data]

    return ToolResult(content=json.dumps(dashboards_data))


def get_all_dashboards_tool() -> Tool[DashboardGetAllSchema]:
    definition = get_tool_definition("dashboards-get-all")

    return Tool(
        name="dashboards-get-all",
        description=definition["description"],
        schema=DashboardGetAllSchema,
        handler=get_all_dashboards_handler,
    )
